### PR TITLE
Log parsed responses

### PR DESCRIPTION
**Pull request by k88hudson**
This is on top of #8  

Logging for DNS query responses, useful for debugging:

image

### CODE REVIEW

File: src/dns-test.js [hunk 1]
```diff
@@ -111,7 +113,36 @@ var dnsAttempts = {
...
+        console.log(m);
+    }
+}
+
+function logDNSResponse(resp, key, transport) {
+    if (!loggingEnabled) {
+        return;
+    }
+    try {
+        let parsed;
```
Comment (ekr) [inline src/dns-test.js:126]: Is it really good form to leave this as |undefined|. Just don't know JS conventions.

File: src/dns-test.js [hunk 2]
```diff
@@ -111,7 +113,36 @@ var dnsAttempts = {
...
+        if (transport === "tcp") {
+            parsed = DNS_PACKET.streamDecode(Buffer.from(resp));
+        } else if (transport === "udp") {
```
Comment (ekr) [inline src/dns-test.js:129]: Seems like a good place for an assert.

File: src/dns-test.js [hunk 3]
```diff
@@ -111,7 +113,36 @@ var dnsAttempts = {
...
+            parsed = DNS_PACKET.decode(Buffer.from(resp));
+        }
+        if (parsed) {
+            console.log(
```
Comment (ekr) [inline src/dns-test.js:133]: Shouldn't this be logMessage, here and below.

File: src/dns-test.js [hunk 4]
```diff
@@ -111,7 +113,36 @@ var dnsAttempts = {
...
+                `DNS Query for ${key}:\n` +
+                `[Q] ${parsed.questions?.map(({name, type}) => `${name} ${type}`).join(",")}\n` +
+                `%c[A] ${parsed.answers?.map(({name, type, data}) => `${name} ${type} ${JSON.stringify(data).slice(0, 50)}`).join("\n    ")} `,
+                'color: green;',
```
Comment (ekr) [inline src/dns-test.js:137]: This is nice compact code, but I think it would be nice to have some kind of ellipsis to indicate that we truncated.

**Review by ekr — APPROVED** [review]
LGTM with changes.

File: src/dns-test.js (hunk 3 above)
Comment (k88hudson) [reply src/dns-test.js:133]: I already have an early return based on `loggingEnabled` at the top, but I suppose I could refactor `logMessage` to handle multiple arguments (which I need here)

File: src/dns-test.js (hunk 2 above)
Comment (k88hudson) [reply src/dns-test.js:104]: In this case, we also want to handle the case where `transport` is not either of these, and just log the raw response (like for the control query)

File: src/dns-test.js (hunk 3 above)
Comment (k88hudson) [reply src/dns-test.js]: Did that

**Comment by ekr** [comment]
LGTM

**Pull request merged**